dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hdfs/ScanContext.java

```diff
@@ -116,6 +130,16 @@ void endWalkDir(FileStatus dir, long nFiles, long nDirs, long accumFileSize) thr
     }
   }
 
+  private void logProgress() {
+    long now = System.currentTimeMillis();
+    if (now - lastLogTime > LOG_PROGRESS_DELAY.toMillis()) {
+      lastLogTime = now;
+      long percentFinished =
+          Longs.constrainToRange(this.numDirsWalked * 100 / totalDirectoryCount, 0, 99);
+      partialProgressLogger.logProgress("HDFS scan " + percentFinished + "% completed");
```

Comment:
A lot of this looks very similar to the shared/common code in [Concurrent]RecordProgressMonitor. Would it be possible to use that?